tools/python/boutiques/publisher.py

```diff
@@ -24,356 +24,179 @@
-from github import Github
-import git
 import json
+import requests
 import os
-import sys
-try:
-    # For Python 3.0 and later
-    from urllib.request import urlopen, URLError
-except ImportError:
-    # Fall back to Python 2's urllib2
-    from urllib2 import urlopen, URLError
 
 
-class Publisher():
-
-    def __init__(self, boutiques_dir, remote_name, tool_author, tool_url, inter,
-                 neurolinks_repo_url, neurolinks_dest_path,
-                 github_user, github_password, no_github):
-        self.boutiques_dir = os.path.abspath(boutiques_dir)
-        self.remote_name = remote_name if remote_name else 'origin'
-        try:
-            self.boutiques_repo = Repo(self.boutiques_dir)
-        except git.exc.InvalidGitRepositoryError as e:
-            # We won't publish a descriptor which is not in a Git repo!
-            sys.stderr.write('error: %s is not a Git repository.\n'
-                             % self.boutiques_dir)
-            sys.exit(1)
-
-        for remote in self.boutiques_repo.remotes:
-            if remote.name == self.remote_name:
-                self.boutiques_remote = remote
-
-        url = self.boutiques_remote.url
-        # If remote URL is on Github, try to guess the URL
-        # of the Boutiques descriptor
-        if "github.com" in url:
-            url = url.replace(".git", "").replace("git@github.com:",
-                                                  "https://github.com/")
-            url = url.replace("https://github.com/",
-                              "https://raw.githubusercontent.com/")
-            url = url.replace("http://github.com/",
-                              "https://raw.githubusercontent.com/")
-            url += "master"
-        self.base_url = url
+class ZenodoError(Exception):
+    pass
 
-        # Try to guess the tool author
-        self.tool_author = tool_author
 
-        self.tool_url = tool_url if tool_url else "http://example.com"
-        self.inter = inter
-        self.no_github = no_github
+class Publisher():
 
-        if self.no_github:
-            return
+    def __init__(self, descriptor_file_name,
+                 creator, affiliation, verbose, sandbox, no_int,
+                 auth_token):
+        # Straightforward assignments
+        self.verbose = verbose
+        self.sandbox = sandbox
+        self.descriptor_file_name = descriptor_file_name
+        self.creator = creator
+        self.affiliation = affiliation
+        self.no_int = no_int
+        self.zenodo_access_token = auth_token
+
+        # Validate and load descriptor
+        validate_descriptor(descriptor_file_name)
+        self.descriptor = json.loads(open(self.descriptor_file_name).read())
+
+        # Fix Zenodo access token
+        if self.zenodo_access_token is None:
+            self.config_file = os.path.join(os.getenv("HOME"), ".boutiques")
+            self.zenodo_access_token = self.get_zenodo_access_token()
+            self.save_zenodo_access_token()
```

Comment:
I though it would be really nice to save the token by default when it's used the first time, so they never need to use it again (or if they use a new one it updates). Accordingly, I updated lines `55` onward to be:

```
self.config_file = os.path.join(os.getenv("HOME"), ".boutiques")
# Fix Zenodo access token
if self.zenodo_access_token is None:
    self.zenodo_access_token = self.get_zenodo_access_token()
self.save_zenodo_access_token()
```